python/ray/tune/result_grid.py

```diff
@@ -148,9 +193,32 @@ def get_dataframe(
     def __len__(self) -> int:
         return len(self._experiment_analysis.trials)
 
-    def __getitem__(self, i) -> Result:
+    def __getitem__(self, i: int) -> Result:
         """Returns the i'th result in the grid."""
-        return self._trial_to_result(self._experiment_analysis.trials[i])
+        return self.get(i)
+
+    def get(
+        self, i: int, *, checkpointing_config: Union[bool, "CheckpointingConfig"] = True
```

Comment:
Do we want to expose this to users? What's the meaning of the ith result? The ordering of the grid is arbitrary right?